backend/app/endpoints.py

```diff
@@ -54,10 +55,13 @@ def from_treasury_with_assets(
         assets = {
             a.token_symbol: PortfolioAsset(
                 allocation=a.balance / treasury.usd_total,
-                volatility=histprices[a.token_symbol]["std_dev"].mean(),
+                volatility=None
+                if histprices[a.token_symbol]["std_dev"].mean() is NaN
+                else histprices[a.token_symbol]["std_dev"].mean(),
                 risk_contribution=a.risk_contribution,
             )
             for a in treasury.assets
+            if hasattr(a, "risk_contribution")
```

Comment:
This line will result in assets not having risk contributions to be destroyed from the dataset. The code in the endpoint should stay as dumb as possible and make no assumptions about the data, just reformat it maybe.

It seems to me can either fill the missing quote price (eg for OWL hist price dataset) with dummy values, either we need to modify the `build_treasury_with_assets` so that it filters out the missing hist prices in a clean manner (see how the `token_symbols_and_addresses` variable is used).

Reply:
You're right in that the consequences of this being quite high... In this case, OWL would not have any metrics when the timeframes were <= 3 months. So this is similar to my comments made below; this seems to only occur when there are no historical price data from coingecko.

Not sure if this is similar to what you're suggesting, but maybe instead we can do
```python
risk_contribution=a.risk_contribution if hasattr(a, "risk_contribution") else None
```

Then a response on `0x0da0c3e52c977ed3cbc641ff02dd271c3ed55afe` (the Gnosis treasury in question) would be
```json
{"assets":{"OWL":{"allocation":0.755362170197285,"volatility":null,"riskContribution":null},"COW":{"allocation":0.23265502725792364,"volatility":0.050341402726052586,"riskContribution":0.9770769110485784},"FRAX":{"allocation":0.003873563610139075,"volatility":0.009174439874320471,"riskContribution":0.0009912899806233883},"STAKE":{"allocation":0.0024757787675371325,"volatility":0.03578447105172835,"riskContribution":0.008246006629808995},"BANK":{"allocation":0.0013547420020908723,"volatility":0.07399246114794918,"riskContribution":0.008709236683074049},"stETH":{"allocation":0.0013526334026573322,"volatility":0.031684742707981044,"riskContribution":0.0025519815492958983},"1INCH":{"allocation":0.0011613184274932398,"volatility":0.04318834257428984,"riskContribution":0.0006694940879545731},"WBTC":{"allocation":0.0008119153938186707,"volatility":0.01680886628260304,"riskContribution":0.0009733157584797315},"LUSD":{"allocation":0.0006084206039240816,"volatility":0.005388326575856505,"riskContribution":9.344931800878378e-05},"aWETH":{"allocation":0.0001863562017383302,"volatility":0.039679948484418255,"riskContribution":0.00040973612705091694},"LQTY":{"allocation":9.679695985363706e-05,"volatility":0.03932349067653369,"riskContribution":0.00015379950050158542},"WETH":{"allocation":2.6506666571796965e-05,"volatility":0.031201562422204638,"riskContribution":5.369827761775218e-05},"MVDG":{"allocation":2.392222921393913e-05,"volatility":0.05171386897453517,"riskContribution":4.94906343229769e-05},"rETH2":{"allocation":1.0848279753121994e-05,"volatility":0.03178057222131695,"riskContribution":2.159040468268637e-05}},"kpis":{"totalValue":437617.49402100005,"volatility":0.032037974135281706,"returnVsMarket":0.003396780661722165},"data":{"2022-07-01":-20618540.279038154,"2022-07-02":-20902534.984683298,"2022-07-03":-20912371.324558128,"2022-07-04":-21069472.545682807,"2022-07-05":-22675517.835087042,"2022-07-06":-22269978.533989146,"2022-07-07":-23348653.803821042}}
```

Reply:
I'd feel more comfortable with this version for now indeed, then we should create an issue for fixing 3 months problem.